python/lsst/pipe/base/graph/graph.py

```diff
@@ -0,0 +1,779 @@
+        """Returns all the `Quantum` associated with a `TaskDef`
+
+        Parameters
+        ----------
+        taskDef : `TaskDef`
+            The `TaskDef` for which `Quantum` are to be queried
+
+        Returns
+        -------
+        set of `Quantum`
+            The `set` of `Quantum` that is associated with the specified
+            `TaskDef`.
+        """
+        return set(self._quanta[taskDef])
+
+    def tasksWithInput(self, datasetTypeName: DSTypeName) -> List[TaskDef]:
+        """Finds all tasks that have the specified dataset type name as an
+        input.
+
+        Parameters
+        ----------
+        datasetTypeName : `DSTypeName`
+            A DSTypeName representing the name of a dataset type to be queried
+
+        Returns
+        -------
+        list of `TaskDef`
+            `TaskDef`s that have the specified `DSTypeName` as an input, list
+            will be empty if no tasks use specified `DSTypeName` as an input.
+
+        Raises
+        ------
+        `KeyError`
+            If the `DSTypeName` is not part of the `QuantumGraph`
+        """
+        return [c for c in self._datasetDict[datasetTypeName].inputs]
+
+    def taskWithOutput(self, datasetTypeName: DSTypeName) -> Optional[TaskDef]:
+        """Finds all tasks that have the specified dataset type name as an
+        output.
+
+        Parameters
+        ----------
+        datasetTypeName : `DSTypeName`
+            A DSTypeName representing the name of a dataset type to be queried
+
+        Returns
+        -------
+        `TaskDef` or None
+            `TaskDef` that outputs `DSTypeName` as an output or None if none of
+            the tasks produce this `DSTypeName`.
+
+        Raises
+        ------
+        `KeyError`
+            If the `DSTypeName` is not part of the `QuantumGraph`
+        """
+        return self._datasetDict[datasetTypeName].output.first
+
+    def tasksWithDSType(self, datasetTypeName: DSTypeName) -> List[TaskDef]:
+        """Finds all tasks that are associated with the specified dataset type
+        name.
+
+        Parameters
+        ----------
+        datasetTypeName : `DSTypeName`
+            A DSTypeName representing the name of a dataset type to be queried
+
+        Returns
+        -------
+        list of `TaskDef`
+            `TaskDef`s that are associated with the specified `DSTypeName`
+
+        Raises
+        ------
+        `KeyError`
+            If the `DSTypeName` is not part of the `QuantumGraph`
+        """
+        results = self.tasksWithInput(datasetTypeName)
+        output = self.taskWithOutput(datasetTypeName)
+        if output is not None:
+            results.append(output)
+        return results
+
+    def findTaskDefByName(self, taskName: str) -> List[TaskDef]:
+        """Determines which `TaskDef`s in this graph are associated with a `str`
+        representing a tasks name (looks at the taskName property of
+        `TaskDefs`).
```

Comment:
`TaskDefs` is plural and won't render properly.  FWIW, I'm not sure
```
`TaskDef`s
```
(which I've seen a few times as well) will either.

Reply:
I think it's
```
`TaskDef`\ s
```